episodes/04-intro-to-visualisation.Rmd

```diff
@@ -0,0 +1,228 @@
+
+First, lets plot the distribution of life expectancy in the `gapminder` dataset:
+
+```{r ggplot}
+  ggplot(data = gapminder,  aes(x = lifeExp) ) + # aesthetics layer 
+  geom_histogram() # geometry layer
+
+```
+
+You can see that in `ggplot` you use `+` as a pipe, to add layers.
+Within the `ggplot()` call, it is the only pipe that will work. But, it is
+possible to chain operations on a data set with a pipe that we have
+already learned: `%>%` ( or `|>`) and follow them by ggplot syntax.
+
+Let's create another plot, this time only on a subset of observations:
+
+```{r ggplot-col}
+gapminder %>%  # we select a data set
+  filter(year == 2007 & 
+         continent == 'Americas') %>%  # and filter it to keep only one year and one continent
+  ggplot(aes(x = country, y = gdpPercap)) +  # the x and y axes represent values of columns
+  geom_col()  # we select a column graph as a geometry
+```
+
+Now, you can iteratively improve how the plot looks like. For example,
+you might want to flip it, to better display the labels.
+
+```{r ggplot-coord-flip}
+gapminder %>%  
+  filter(year == 2007, 
+         continent == 'Americas') %>% 
+  ggplot(aes(x = country, y = gdpPercap)) + 
+  geom_col()+ 
+  coord_flip()  # flip axes
+```
+
+One thing you might want to change here is the order in which countries
+are displayed. It would be easier to compare GDP per capita, if they
+were showed in order. To do that, we need to reorder factor levels (you
+remember, we've already done this before).
+
+Now the order of the levels will depend on another variable - GDP per
+capita.
+
+```{r ggplot-color}
+gapminder %>%  
+  filter(year == 2007, 
+         continent == 'Americas') %>% 
+  mutate(country = fct_reorder(country, gdpPercap )) %>%  # reorder factor levels
+  ggplot(aes(x = country , y = gdpPercap)) + 
+  geom_col() +
+  coord_flip()
+
+```
+
+Let's make things more colourful - let's represent the average life
+expectancy of a country by colour
+
+```{r ggplot-colors}
+gapminder %>%  
+  filter(year == 2007, 
+         continent == 'Americas') %>% 
+  mutate(country = fct_reorder(country, gdpPercap )) %>%
+  ggplot(aes(x = country, y = gdpPercap, fill = lifeExp  )) +  # fill argument for colouring surfaces, colour for points and lines
+  geom_col()+ 
+  coord_flip()
+
+
+```
+
+We can also adapt the colour scale. Common choice that is used for its
+readability and colorblind-proofness are the palettes available in the
+`viridis` package.
+
+```{r ggplot-colors-adapt}
+gapminder %>%  
+  filter(year == 2007, 
+         continent == 'Americas') %>% 
+  mutate(country = fct_reorder(country, gdpPercap )) %>%
+  ggplot(aes(x = country, y = gdpPercap, fill = lifeExp   )) + 
+  geom_col()+ 
+  coord_flip()+
+  scale_fill_viridis_c()  # _c stands for continuous scale
+
+```
+
+Maybe we don't need that much information about the life expectancy. We
+only want to know if it's below or above average.
```

Comment:
```suggestion
only want to know if it's below or above average. We will make use of the `if_else()` function inside `mutate()` to create a new column `lifeExpCat` with the value `high` if life expectancy is above average and `low` otherwise. Note the usage of the `if_else()` function: `if_else(<condition>, <value if TRUE>, <value if FALSE>)`.
```